packages/core/src/browser/label-provider.ts

```diff
@@ -47,10 +50,21 @@ export const FOLDER_ICON = DEFAULT_FOLDER_ICON;
 export const FILE_ICON = DEFAULT_FILE_ICON;
 
 export const LabelProviderContribution = Symbol('LabelProviderContribution');
+/**
+ * A {@link LabelProviderContribution} determines how specific elemens/nodes are displayed in the workbench.
+ * Theia views use a common {@link LabelProvider} to determine the label and/or an icon for elements shown in the UI. This includes elements in lists
+ * and trees, but also view specific locations like headers. The common {@link LabelProvider} collects all {@links LabelProviderContribution} and delegates
+ * to the contribution with the highest priority. This is determined via calling the {@link LabelProviderContribution.canHandle} function, so contributions
+ * define which elements they are responsible for.
+ * As an example, a {@links LabelProviderContribution} can contribute an icon for a specific file type. This would be used in multiple locations displaying files,
+ * e.g. the file explorer, the header of editors or in file dialogues. Therefore, as arbitrary views can consume LabelProviderContributions, they must be generic
+ * for the covered element type, not view specific. Label providers and contributions can be used for arbitrary element and node types, e.g. for markers or domain-specific
+ * elements.
+ */
```

Comment:
I believe the documentation can be simplified greatly, as is will lead to it being outdated.
- There is no need to speak about the internal details of implementations of the interface (downstream apps can provide their own implementation if desired), and describing a possible implementation for the interface can lead to it being outdated quickly. 
- There is no need to mention uses of the interface (ex: lists, tree, headers) as they are not the only possible options.
- Note sure the example is necessary? An example would be more an implementation of the interface developers can view instead.